tardis/rest/app/security.py

```diff
@@ -1,127 +1,119 @@
+import secrets
 from ...configuration.configuration import Configuration
 from ...exceptions.tardisexceptions import TardisError
 
 from bcrypt import checkpw, gensalt, hashpw
-from fastapi import Depends, HTTPException, status
-from fastapi.security import OAuth2PasswordBearer, SecurityScopes
-from jose import JWTError, jwt
-from pydantic import BaseModel, ValidationError
+from fastapi import HTTPException, status, Depends
+
+from fastapi.security import SecurityScopes
+
+from pydantic import BaseModel
+from fastapi_jwt_auth import AuthJWT
 
-from datetime import datetime, timedelta
 from functools import lru_cache
-from typing import List, Optional
+from typing import Optional, List
 
 
-class TokenData(BaseModel):
-    user_name: Optional[str] = None
-    scopes: List[str] = []
+class Settings(BaseModel):
+    authjwt_secret_key: str = secrets.token_hex(128)
+    authjwt_token_location: set = {"cookies"}
+    authjwt_cookie_samesite: str = "strict"
+    # TODO: change this to true in production so only https traffic is allowed
+    authjwt_cookie_secure: bool = False
+    # TODO: Set this too to True. But as soon as possible.
+    # Update: As 'same_site' is strict this is probably enough.
+    authjwt_cookie_csrf_protect: bool = False
 
 
-class UserCredentials(BaseModel):
-    user_name: str
-    hashed_password: str
-    scopes: List[str]
+@AuthJWT.load_config
+def get_config():
+    return Settings()
```

Comment:
IMHO the configuration should be adjustable from the `TARDIS` configuration of the REST API, with reasonable defaults.

Reply:
Yes, exactly. Would be reasonable depending on the setup. development vs. production. Frontend proxy + backend or just one web server.

Reply:
I'm having issues with implementing this. `Configuration().Services` throws an `AttributeError` when being called from `get_config()`. I guess it gets initialized after the file runs, but I'm having difficulty understanding when it's ready to be used.